### ISSUE TITLE
Support ESLint 8.x

### ISSUE THREAD

**Issue by MichaelDeBoey** [microsoft/rushstack#2890]
ESLint v8.0.0 is [released](https://eslint.org/blog/2021/10/eslint-v8.0.0-released) 🎉

It would be awesome to have official ESLint 8 support. 👊
I'm happy to help where I can of course 🙂

**Comment by iclanton** [comment]
@octogonz - want to take a look at this?

**Comment by octogonz** [comment]
We don't normally adopt or support beta versions of tools (since it's already a nontrivial time investment to keep up with officially released versions 😇).

I'm fine however if someone wants to make a PR that enables usage of `eslint-v8.0.0-beta-*`.  Heft should probably print a notice to the console saying it's not officially supported.

I'm also curious to see what will happen with [@rushstack/eslint-patch](https://www.npmjs.com/package/@rushstack/eslint-patch).

**Comment by D4N14L** [comment]
@marcoreni is correct, this should be re-opened. I will take a look at this soon.

**Comment by lukashass** [comment]
@D4N14L FYI I already started this in #3009. Not sure if that covers everything needed to close this issue. Also, it is blocked by upstream changes as far as I can tell, but feel free to continue.

**Comment by D4N14L** [comment]
@lukashass yes, I noticed this just recently, sorry for missing that! I already had some additional changes in my branch, so I'll likely create a separate PR for it. I'll tag you when I get that out.

**Comment by stefaneg** [comment]
I am working on getting eslint to work with heft, having trouble making it work with eslint 8.7.0.

Is it possible that the documentation is out of date here?

https://www.npmjs.com/package/@rushstack/eslint-config

**Comment by D4N14L** [comment]
@stefaneg it seems like it works successfully (see this PR: #3185). What is the issue you're having?

**Comment by stefaneg** [comment]
The error I get is this when execute heft build with .eslintrc.js exactly as it specified in the documentation. I have verified that everything is present in node_modules at the correct paths.

```
> heft build 
Project build folder is "/Users/gulli/src/github.com/rush-example/libraries/logging"
Starting build
 ---- Pre-compile started ----
 ---- Pre-compile finished (1ms) ----
 ---- Compile started ----
[typescript] Using TypeScript version 3.9.10
[eslint] Using ESLint version 8.7.0
 ---- Compile encountered an error (2239ms) ----
-------------------- Finished (2.51s) --------------------
Project: @tm.is/logging@3.6.10
Heft version: 0.44.2
Node version: v16.13.1
Error: Cannot read config file: /Users/gulli/src/github.com/rush-example/libraries/logging/.eslintrc.js
Error: Failed to patch ESLint because the calling module was not recognized
```

I tried downgrading ESLint to 7.32.0, and the error persists, so the issue seems to be something else.

**Comment by stefaneg** [comment]
It looks like the message is originating here:

https://github.com/microsoft/rushstack/blob/0177df04f58da07b52bd7af280012f18eefa841d/eslint/eslint-patch/src/modern-module-resolution.ts#L158

Hard to be sure unless there is an output processor that outputs only the first line of the message to stderr.
